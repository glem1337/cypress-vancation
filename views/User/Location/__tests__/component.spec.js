import { shallow } from 'enzyme';

import UserLocation from '../component';

describe('UserLocation component matches snapshot', () => {
  const defaultProps = {
    user: {
      firstName: 'Johnie',
      lastName: 'Doe',
      country: 'Ukraine',
      city: 'Dnepr',
      email: 'test@example.com',
      about: 'Manager',
    },
  };

  it('with default props', () => {
    const wrapper = shallow(<UserLocation {...defaultProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('with editLink', () => {
    const props = {
      ...defaultProps,
      editLink: <a>Edit Link</a>,
    };
    const wrapper = shallow(<UserLocation {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
