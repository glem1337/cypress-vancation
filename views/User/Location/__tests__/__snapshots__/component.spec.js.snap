// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserLocation component matches snapshot with default props 1`] = `
<div
  className="employee-profile__card-section"
  data-cy="profile-location"
>
  <p
    className="employee-profile__card-section-title"
  >
    <span
      className="mr-8"
    >
      <Memo(FormattedMessage)
        id="user.location"
      />
    </span>
  </p>
  <p
    className="employee-profile__text mb-8"
  >
    <Memo(FormattedMessage)
      id="user.country"
      values={
        Object {
          "country": "Ukraine",
        }
      }
    />
  </p>
  <p
    className="employee-profile__text"
  >
    <Memo(FormattedMessage)
      id="user.city"
      values={
        Object {
          "city": "Dnepr",
        }
      }
    />
  </p>
</div>
`;

exports[`UserLocation component matches snapshot with editLink 1`] = `
<div
  className="employee-profile__card-section"
  data-cy="profile-location"
>
  <p
    className="employee-profile__card-section-title"
  >
    <span
      className="mr-8"
    >
      <Memo(FormattedMessage)
        id="user.location"
      />
    </span>
    <a>
      Edit Link
    </a>
  </p>
  <p
    className="employee-profile__text mb-8"
  >
    <Memo(FormattedMessage)
      id="user.country"
      values={
        Object {
          "country": "Ukraine",
        }
      }
    />
  </p>
  <p
    className="employee-profile__text"
  >
    <Memo(FormattedMessage)
      id="user.city"
      values={
        Object {
          "city": "Dnepr",
        }
      }
    />
  </p>
</div>
`;
